coderd/workspaceagents_test.go

```diff
@@ -876,6 +877,60 @@ func TestWorkspaceAgentReportStats(t *testing.T) {
 			"%s is not after %s", newWorkspace.LastUsedAt, r.Workspace.LastUsedAt,
 		)
 	})
+
+	t.Run("FailDeleted", func(t *testing.T) {
+		t.Parallel()
+
+		client, db := coderdtest.NewWithDatabase(t, nil)
+		user := coderdtest.CreateFirstUser(t, client)
```

Comment:
We are trying not to use owner for tests lately. This also helps for the `dbauthz.As` part.
```suggestion
		owner, db := coderdtest.NewWithDatabase(t, nil)
		ownerUser := coderdtest.CreateFirstUser(t, owner)
		client, admin := coderdtest.CreateAnotherUser(t, owner, ownerUser.OrganizationID, rbac.RoleTemplateAdmin(), rbac.RoleUserAdmin())
```